src/factorials.jl

```diff
@@ -95,14 +95,42 @@ end
 """
     multinomial(k...)
 
-Multinomial coefficient where `n = sum(k)`.
+Compute the multinomial coefficient
+``\\binom{n}{k_1,k_2,...,k_i} = \\frac{n!}{k_1!k_2! \\cdots k_i!}, n = \\sum{k_i}``.
```

Comment:
This pr mainly addresses overflow issues, perhaps I should split the document and update it to another pr?

![image](https://github.com/user-attachments/assets/12374202-3746-4822-bba6-55cd92b8eeb1)